### ISSUE TITLE
Code action keybinding isn't available in visual mode

### ISSUE THREAD

**Issue by byrmsh** [nvim-lua/kickstart.nvim#1060]
## Describe the bug

Code action keybinding `vim.lsp.buf.code_action` is not available in visual mode. This prevents using refactoring functionality such as "extract selected code into function". Is this intentional? If it is not, I can quickly send a PR. I searched for this on the issues and pull requests but couldn't find anything matching.

## To Reproduce

1. Select any code block in visual mode in a buffer with an LSP client running
2. Use default keybinding `<leader>ca` for code action
3. Selected code is deleted and mode is set to insert mode

## Desktop

- OS: Linux
- Terminal: Alacritty

## Neovim Version

```
NVIM v0.10.0
Build type: RelWithDebInfo
LuaJIT 2.1.1713773202
Run "nvim -V1 -v" for more info
```

**Comment by iton0** [comment]
Yeah make a PR for this

**Comment by VlaDexa** [comment]
> 3\. Selected code is deleted and mode is set to insert mode

That's the default behavior of `c` keybind.

Refactoring actions are a bit broken in nvim as of my test, they don't remove the selection and don't offer an automatic way to change function name when you extract into a function. If you seek a bit more proper support I recommend looking into [refactoring.nvim](https://github.com/ThePrimeagen/refactoring.nvim).

**Comment by byrmsh** [comment]
> That's the default behavior of `c` keybind.

Yeah I am aware of this as it is one of the essential keybindings of Vim/Neovim. I just wanted to write the steps in detail.

> Refactoring actions are a bit broken in nvim as of my test, they don't remove the selection and don't offer an automatic way to change function name when you extract into a function. If you seek a bit more proper support I recommend looking into [refactoring.nvim](https://github.com/ThePrimeagen/refactoring.nvim).

I already tried that but my experience was actually simply better without refactoring.nvim. After realizing default kickstart.nvim config doesn't allow running code actions in visual mode, I tried setting up refactoring.nvim on a freshly cloned, up-to-date kickstart.nvim config. However, I encountered buggy behavior and straight up crashes during selection of refactors and afterwards, making it overall unstable for me. Instead of spending time on investigating the crashes caused by this extra plugin, I just simply removed it and updated kickstart.nvim's default code action keybinding to work in visual mode. This simply worked flawlessly, providing the typical refactoring experience VSCode offered, at least for the Go project I tested it on, without installing yet another plugin. Also, I haven't experienced the issue you mentioned about selection removal and I don't mind renaming the newly created extracted function.
